packages/wp-story-editor/src/components/metaBoxes/metaBoxes.js

```diff
@@ -13,29 +13,68 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 /**
  * External dependencies
  */
 import styled from 'styled-components';
-import { useEffect } from '@web-stories-wp/react';
+import {
+  useEffect,
+  useRef,
+  useState,
+  createPortal,
+} from '@web-stories-wp/react';
+import { useConfig, useStory } from '@web-stories-wp/story-editor';
+/**
+ * Internal dependencies
+ */
+import MenuItem from './menuItem';
 
 /**
  * Internal dependencies
  */
-import { useConfig } from '../../../app/config';
 import MetaBoxesArea from './metaBoxesArea';
 import useMetaBoxes from './useMetaBoxes';
 
 const Wrapper = styled.div``;
 
+const Area = styled.div`
+  grid-area: ${({ area }) => area};
+  position: relative;
+  overflow: hidden;
+  z-index: 2;
+`;
+
+const MetaBoxesContainer = styled(Area).attrs({
+  area: 'supplementary',
+})`
+  overflow-y: auto;
+`;
+
 function MetaBoxes() {
   const { metaBoxesVisible, hasMetaBoxes } = useMetaBoxes(({ state }) => ({
     hasMetaBoxes: state.hasMetaBoxes,
     metaBoxesVisible: state.metaBoxesVisible,
   }));
-
+  const [showMenuButton, updateMenuButtonState] = useState(false);
+  const menuButtonContainer = useRef(null);
   const { postType, metaBoxes = {} } = useConfig();
+  const { pages } = useStory(({ state: { pages } }) => ({ pages }));
+
+  useEffect(() => {
+    let timeout = null;
+
+    if (hasMetaBoxes && pages.length >= 0) {
```

Comment:
Did you perhaps mean `pages.length > 0`?

If length is `0` it means the story hasn't fully loaded in the editor yet.